File: 1108/c#/1108.cs
// mcs -out:1108.exe 1108.cs
// mono 1108.exe

using System;
using System.Text.RegularExpressions;

public class Solution
{
    public string DefangIPaddr(string address)
    {
        string result = Regex.Replace(address,@"(\.)","[.]");

        return result;
    }

    static void Main(string[] args)
    {
        Solution solution = new Solution();
        Console
            .WriteLine("[{0}]",
            string
                .Join(", ",
                solution
                    .DefangIPaddr("1.1.1.1")));

        Console
            .WriteLine("[{0}]",
            string
                .Join(", ",
                solution
                    .DefangIPaddr("255.100.50.0")));
    }
}

File: 1108/typescript/1108.ts
// tsc 1108.ts
// node 1108.js

function defangIPaddr(address: string): string {
    let out: string = address.replace(/\./g, "[.]");
    return out;
};

console.log(defangIPaddr("1.1.1.1"));
console.log(defangIPaddr("255.100.50.0"));
